src/provider/application.js

```diff
@@ -72,7 +73,32 @@ class Application extends EventEmitter {
       const width = calculateWidth(this.resizeConfig.WidthCalculationMethod);
       this.JSONRPC.notification('resize', [null, `${width}px`]);
     } else {
-      const height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+      let height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+
+      // If targetSelectors is specified from Provider or Consumer or both,
+      // need to calculate the height based on specified target selectors
+      if (this.targetSelectors.length > 0 || this.resizeConfig.targetSelectors) {
+        let targetSelectors = [];
+
+        if (this.targetSelectors.length > 0) {
+          targetSelectors = targetSelectors.concat(this.targetSelectors);
+        }
+        if (this.resizeConfig.targetSelectors) {
+          targetSelectors = targetSelectors.concat(this.resizeConfig.targetSelectors);
+        }
+
+        // Remove duplicates
+        targetSelectors = targetSelectors.filter(
+          (elem, index, arr) => arr.indexOf(elem) === index
```

Comment:
this probably isn't necessary if we make a single `querySelectorAll` call.. it likely does this in its implementation